### PR TITLE
binary, curried composition

### PR DESCRIPTION
**Pull request by buzzdecafe**
anyone interested in a stricter `compose` function?

### CODE REVIEW

File: src/o.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ *
+ *      var classyGreeting = name => "The name's " + name.last + ", " + name.first + " " + lastName
+ *      var yellGreeting = R.o(R.toUpper, classyGreeting);
+ *      yellGreeting({first: 'James', last: 'Bond'}); //=> "THE NAME'S BOND, JAMES BOND"
+ *
+ *      R.o(Math.abs, R.multiply(2))(-4) //=> 8
+ *
+ * @symb R.o(f, g)(x) = f(g(x))
+ */
+module.exports = _curry2(function o(f, g) {
```
Comment (davidchambers) [inline src/o.js]: Why not `_curry3(function o(f, g, x) { ... })`?

Comment (buzzdecafe) [inline src/o.js]: no good reason

File: src/o.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ * @param {Function} g
+ * @return {Function}
+ * @see R.compose, R.pipe
+ * @example
+ *
+ *      var classyGreeting = name => "The name's " + name.last + ", " + name.first + " " + lastName
+ *      var yellGreeting = R.o(R.toUpper, classyGreeting);
+ *      yellGreeting({first: 'James', last: 'Bond'}); //=> "THE NAME'S BOND, JAMES BOND"
+ *
+ *      R.o(Math.abs, R.multiply(2))(-4) //=> 8
```
Comment (davidchambers) [inline src/o.js]: You should choose an example which demonstrates the order in which the functions are applied.

**Comment by CrossEye** [comment]
:herb:, except that, even though I understand the reason for `'o'`, I'm not a real fan of Ramda using such one-character names.  Even `R.T` and `R.F` bother me.

**Comment by davidchambers** [comment]
The right name for this function is `compose`. ;)

File: src/o.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+ * @see R.compose, R.pipe
+ * @example
+ *
+ *      var classyGreeting = name => "The name's " + name.last + ", " + name.first + " " + lastName
+ *      var yellGreeting = R.o(R.toUpper, classyGreeting);
+ *      yellGreeting({first: 'James', last: 'Bond'}); //=> "THE NAME'S BOND, JAMES BOND"
+ *
+ *      R.o(R.multiply(10), R.add(10))(-4) //=> 60
+ *
+ * @symb R.o(f, g)(x) = f(g(x))
```
Comment (davidchambers) [inline src/o.js]: Should we use `R.o(f, g, x)` here?

**Comment by CrossEye** [comment]
Or `B`.  

Or `bluebird`.  

Perhaps we could call it `supercalifragilisticexpialidocious`.

File: src/o.js (hunk 3 above)
Comment (buzzdecafe) [inline src/o.js]: yes, well spotted

**Comment by buzzdecafe** [comment]
> The right name for this function is `compose`. ;)

:fire:

ok i think this one is good to go; please give it a final look & merge.

**Comment by CrossEye** [comment]
@buzzdecafe:
> ok i think this one is good to go; please give it a final look & merge.

You really want to go with the name `o`?

**Comment by buzzdecafe** [comment]
> You really want to go with the name o?

i agree with @davidchambers that i'd rather see this called `compose` and the other get renamed to `comp` [a la clojure](https://clojuredocs.org/clojure.core/comp), which it resembles; or something else more descriptive, e.g. `variadicCompose`, `composeVar` etc.

Given that the "right" name is squatted on atm, naming the function for the composition operator seems good to me. It also has the advantage of concision for what ought to be one of the primary functions of the library.

**Comment by CrossEye** [comment]
Merging under protest.  :smile:

**Pull request merged**